commands/uploader.go

```diff
@@ -377,6 +362,24 @@ func getVerifyStateFor(endpoint lfsapi.Endpoint) verifyState {
 	return verifyStateDisabled
 }
 
+// supportsLockingAPI returns whether or not a given lfsapi.Endpoint "e"
+// is known to support the LFS locking API by whether or not its hostname is
+// included in the list above.
+func supportsLockingAPI(e lfsapi.Endpoint) bool {
+	u, err := url.Parse(e.Url)
+	if err != nil {
+		tracerx.Printf("commands: unable to parse %q to determine locking support: %v", e.Url, err)
+		return false
+	}
+
+	for _, host := range hostsWithKnownLockingSupport {
```

Comment:
Good call. I think as we add/get support among more hosts, we should have the flexibility to specify by scheme, path prefix, and etc. This may be premature, but I changed the `hostsWithKnownLockingSupport` to be a `[]*url.URL`, where we can hold onto and compare against fields like `Host`, `Scheme`, Port()`, and etc.

That change is in https://github.com/git-lfs/git-lfs/pull/2162/commits/739bd779b35aa1a13b6572219c4ed3fc6dc9738e. I wrote a test case on the Playground [here](https://play.golang.org/p/o6biD__ny6), which I turned into a unit test in https://github.com/git-lfs/git-lfs/pull/2162/commits/a6c7b965c76147f91718aa0448359507b2142301.